packages/propel/src/charts/scatter-chart/root.tsx

```
@@ -0,0 +1,149 @@
+            y: 10,
+        },
+        legend,
+        showTooltip = true,
+    } = props;
+    // states
+    const [activePoint, setActivePoint] = useState<string | null>(null);
+    const [activeLegend, setActiveLegend] = useState<string | null>(null);
+    // derived values
+    const itemKeys = useMemo(() => scatterPoints.map((point) => point.key), [scatterPoints]);
+    const itemLabels: Record<string, string> = useMemo(
+        () => scatterPoints.reduce((acc, point) => ({ ...acc, [point.key]: point.label }), {}),
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)